middleware/service_broker_rate_limiter.rb

```diff
@@ -3,6 +3,14 @@
 
 module CloudFoundry
   module Middleware
+    RateLimitEndpoint = Struct.new(:endpoint_pattern, :request_methods)
+
+    RATE_LIMITED_ENDPOINTS = [
+      RateLimitEndpoint.new(%r{\A/v2/(service_instances|service_credential_bindings|service_route_bindings)}, %w[PUT POST DELETE PATCH]),
+      RateLimitEndpoint.new(%r{\A/v3/(service_instances|service_credential_bindings|service_route_bindings)/.+/parameters\z}, %w[GET]),
+      RateLimitEndpoint.new(%r{\A/v3/(service_instances|service_credential_bindings|service_route_bindings)}, %w[PUT])
```

Comment:
Does the CC implement any PUT methods for /v3/service_instances etc? At least the CF API doc doesn't mention it.

For v2, PUT is required, e.g. https://v2-apidocs.cloudfoundry.org/service_instances/update_a_service_instance.html

Reply:
No, you are right, the only place I found  where [PUT is used for v3 is for cancel a task](https://github.com/cloudfoundry/cloud_controller_ng/blob/eece70a64d6ffb8c0b0a6730f58cb255c0543a7f/docs/v3/source/includes/resources/tasks/_cancel.md.erb#L34), but it is also mentioned, that it is deprecated.